STIRConfig.cmake Find Boost package versioning error

Build a project that uses STIR as a package via `find_package(STIR 6.0 REQUIRED CONFIG)` I get the following error:

```
CMake Error at C:/Program Files/JetBrains/CLion 2024.1.5/bin/cmake/win/x64/share/cmake-3.29/Modules/FindBoost.cmake:1654 (message):
  When requesting a specific version of Boost, you must provide at least the
  major and minor version numbers, e.g., 1.34
Call Stack (most recent call first):
  ${STIR_out}/install/x64-Release/lib/cmake/STIR-6.2/STIRConfig.cmake:106 (find_package)
  CMakeLists.txt:49 (find_package)


CMake Error at C:/Program Files/JetBrains/CLion 2024.1.5/bin/cmake/win/x64/share/cmake-3.29/Modules/FindPackageHandleStandardArgs.cmake:230 (message):
  Could NOT find Boost (missing: Boost_INCLUDE_DIR) (Required is at least
  version "108200")
Call Stack (most recent call first):
  C:/Program Files/JetBrains/CLion 2024.1.5/bin/cmake/win/x64/share/cmake-3.29/Modules/FindPackageHandleStandardArgs.cmake:600 (_FPHSA_FAILURE_MESSAGE)
  C:/Program Files/JetBrains/CLion 2024.1.5/bin/cmake/win/x64/share/cmake-3.29/Modules/FindBoost.cmake:2393 (find_package_handle_standard_args)
  ${STIR_out}/install/x64-Release/lib/cmake/STIR-6.2/STIRConfig.cmake:106 (find_package)
  CMakeLists.txt:49 (find_package)


-- Configuring incomplete, errors 
```

It seems the issue stems from https://github.com/UCL/STIR/blob/4953e27e683845ee3967895ec955af09127425eb/src/cmake/STIRConfig.cmake.in#L82
which was introduced by https://github.com/UCL/STIR/pull/1536

The line at `lib/cmake/STIR-6.2/STIRConfig.cmake:106 ` is set to `find_package(Boost 108200 REQUIRED)` which has the wrong version formatting. 

Replacing the line in the installed `STIRConfig.cmake` file with `find_package(Boost 1.82.0 REQUIRED)` allows the project to compile.

It seems the variable in `find_package(Boost @Boost_VERSION@ REQUIRED)` is incorrect.

--- a/src/cmake/STIRConfig.cmake.in
+++ b/src/cmake/STIRConfig.cmake.in
@@ -79,7 +79,7 @@ else()
   SET(STIR_FIND_TYPE "REQUIRED")
 endif()
 
-find_package(Boost @Boost_VERSION@ REQUIRED)
+find_package(Boost @Boost_VERSION_STRING@ REQUIRED)
 
 if (@ITK_FOUND@)
   message(STATUS "ITK support in STIR enabled.")

